instrumentation/otelgrpc: rename Typ to Type (#2688)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

### Changed

> - Rename Typ to Type on `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc#InterceptorInfo` field. (#2688)

Reviewer: ```suggestion
- Rename the `Typ` field of `"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc".InterceptorInfo` to `Type`. (#2688)
```
